widget/date_entry.go

```diff
@@ -154,11 +160,27 @@ func (e *DateEntry) setupDropDown() *Button {
 		e.dropDown = NewCalendar(time.Now(), e.setDate)
 	}
 	dropDownButton := NewButton("", func() {
+		if e.Date != nil && !e.Date.IsZero() {
+			e.dropDown.SetSelectedDate(*e.Date)
+			e.dropDown.SetDisplayedDate(*e.Date)
+		} else {
+			e.dropDown.SetSelectedDate(time.Time{})
+			e.dropDown.SetDisplayedDate(time.Time{})
+		}
+
 		c := fyne.CurrentApp().Driver().CanvasForObject(e.super())
 
-		e.popUp = NewPopUp(e.dropDown, c)
+		e.popUp = NewPopUp(&fyne.Container{
```

Comment:
container.NewVBox is a lot cleaner.

Reply:
Impossible due to import cycle... (container imports widget and widgets imports container...)